sim/riscv/interp.c

```diff
@@ -154,13 +158,23 @@ sim_create_inferior (SIM_DESC sd, struct bfd *abfd,
 {
   SIM_CPU *cpu = STATE_CPU (sd, 0);
   sim_cia addr;
+  Elf_Internal_Phdr *phdr;
+  int i, phnum;
 
   /* Set the PC.  */
   if (abfd != NULL)
     addr = bfd_get_start_address (abfd);
   else
     addr = 0;
   sim_pc_set (cpu, addr);
+  phdr = elf_tdata (abfd)->phdr;
```

Comment:
This assumes the highest address in the executable is the start of heap.  But if you use a linker script, you could have put special sections in high memory, and putting heap after them would be wrong.  Or alternatively, a board could have multiple memory regiions, and the region where the text/data is stored is not necessarily the highest address one.  Normally, one looks for the symbol _end and uses that as the start of heap.  But there is no one right way to do this.